### PR TITLE
Remove ?w=&h= url params for self hosted sites in the MediaAdapter and EditPostSettingsFragment

### PR DESCRIPTION
**Pull request by maxme**

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public void onBindViewHolder(GridViewHolder holder, int position) {
                 if (SiteUtils.isPhotonCapable(mSite)) {
                     thumbUrl = PhotonUtils.getPhotonImageUrl(media.getUrl(), mThumbWidth, mThumbHeight);
                 } else {
-                    thumbUrl = UrlUtils.removeQuery(media.getUrl()) + "?w=" + mThumbWidth + "&h=" + mThumbHeight;
+                    thumbUrl = UrlUtils.removeQuery(media.getUrl());
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java]: I'm very tempted to use `media.getThumbnailUrl()` here instead, thumbnails are 150x150 by default in WordPress, that makes the media grid ugly but images load much faster (Average of 8kb images vs. 2Mb).

**Comment by maxme** [comment]
Ref: https://github.com/wordpress-mobile/WordPress-Android/pull/5841/files#r115000629

Using `getThumbnailUrl`:

screenshot-2017-05-05_15 51 26 407

Without:


screenshot-2017-05-05_15 52 47 828

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 2]
```diff
@@ -431,20 +433,24 @@ public void updateFeaturedImage(long id) {
             if (mFeaturedImageId > 0) {
                 MediaModel media = mMediaStore.getSiteMediaWithId(mSite, mFeaturedImageId);
 
-                if (media == null) {
+                if (media == null || getActivity() == null) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: Should not be `|| !isAdded()` ?

**Comment by daniloercoli** [comment]
> , that makes the media grid ugly but images load much faster (Average of 8kb images vs. 2Mb).

150px wide pictures are pretty small and look ugly :| 
Maybe do this when on the default device network connection (Not on WiFi, not on any other type of network connection).

We should probably detect when Jetpack is not installed on the self hosted site, and advertise you can save data if you install it, and your site has lot of pictures.

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java]: > 150px wide pictures are pretty small and look ugly :|

Yep :/

> Maybe do this when on the default device network connection (Not on WiFi, not on any other type of network connection).

That's a good idea. What do you think @bysl?

> We should probably detect when Jetpack is not installed on the self hosted site, and advertise you can save data if you install it, and your site has lot of pictures.

We can detect it, but I'm not sure how to advertise this without being too much "in your face", not doing that in this PR anyway.

**Comment by nbradbury** [comment]
> 150px wide pictures are pretty small and look ugly :|
> Maybe do this when on the default device network connection (Not on WiFi, not on any other type of network connection).

Another option is to load the thumbnail first, then load the larger image. That way the user sees the images much more quickly but they're not stuck looking at pixelated images for too long. We do something similar in the reader photo viewer [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderPhotoView.java).

**Comment by daniloercoli** [comment]
> Another option is to load the thumbnail first, then load the larger image.

Nice idea, probably better to open a new PR for it.

@maxme, once my comment above is addressed we can merge this PR.

**Comment by maxme** [comment]
> probably better to open a new PR for it.

Yes, even better if we filled a new ticket in core, because we shouldn't download 50 x 5Mb pictures when the user open the media grid.

**Comment by maxme** [comment]
> @maxme, once my comment above is addressed we can merge this PR.

Updated in  	ebab027

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**